ibm/service/cis/resource_ibm_cis_mtls.go

```diff
@@ -0,0 +1,225 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package cis
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	cisMtlsID            = "mtls_id"
+	cisMtlsCert          = "certificate"
+	cisMtlsCertName      = "name"
+	cisMtlsCertID        = "cert_id"
+	cisMtlsHostName      = "associated_hostname"
+	cisMtlsCertCreatedAt = "created_at"
+	cisMtlsCertUpdatedAt = "updated_at"
+	cisMtlsCertExpireOn  = "expires_on"
+)
+
+func ResourceIBMCISMtls() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMCISMtlsCreate,
+		ReadContext:   resourceIBMCISMtlsRead,
+		UpdateContext: resourceIBMCISMtlsUpdate,
+		DeleteContext: resourceIBMCISMtlsDelete,
+		Importer:      &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisMtlsID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Mtls transaction ID",
+			},
+			cisMtlsCert: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Certificate contents",
+			},
+			cisMtlsCertName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Certificate name",
+			},
+			cisMtlsHostName: {
```

Comment:
API is taking a list of string why is it a string here

Reply:
OK. Current implementation can associate a single host only. I'll check it.